server/plugin.go

```diff
@@ -265,3 +266,33 @@ func (p *Plugin) handleAutoCallsRecording(post *model.Post, postingUser *model.U
 
 	return nil
 }
+
+func (p *Plugin) FileWillBeUploaded(c *plugin.Context, info *model.FileInfo, file io.Reader, output io.Writer) (*model.FileInfo, string) {
+	if info.Content != "" {
```

Comment:
This address @streamer45's concern about not regenerating if the file info already have a content. But keep in mind that you have to generate the file info already with the `Content` field properly populated, if not, the plugin is going to try to generate it anyway.

Reply:
I wish we had a `FileInfo.Type` field to handle custom files because as things stand, calls recordings won't ever have the `Content` set as the transcription process would happen independently, after they have been uploaded. So in the best case scenario we'd be patching the file info at a later time (not even sure we can do it without API additions), or use post props.